pkg/compiler/attrs.go

```diff
@@ -50,6 +54,36 @@ func init() {
 			comp.error(attr.Pos, "bad struct %v alignment %v (must be a sane power of 2)", name, a)
 		}
 	}
+	fieldAttrCheckDirConsts := func(comp *compiler, parent ast.Node, attr *ast.Type) {
+		_, typ, _ := parent.Info()
+		switch parent.(type) {
+		case *ast.Field:
+			f := parent.(*ast.Field)
+			attrs := comp.parseAttrs(fieldAttrs, f, f.Attrs)
```

Comment:
This feels wrong.
The CheckConsts callback is called per-attribute and is supposed to check only this single attribute value. But here we re-parse all attributes for each attribute (quadratic) and as the result has this error reporting precedence problem.
I think we need to move this check somewhere else and do it once. checkStructFields may be a good location.

Reply:
Now it is checked if the attribute belongs to a field within per attribute checks ([here](https://github.com/google/syzkaller/pull/2008/commits/82ffbda67776502531b39b560e33ac8956d90827#diff-92355e9c4025b723c29c1967a304dc11R60-R63)).

As suggested, the remaining checks are moved to checkStructFields ([here](https://github.com/google/syzkaller/pull/2008/commits/82ffbda67776502531b39b560e33ac8956d90827#diff-47a4f07e1880cd23bc05dca73d329c1dR172-R181)).

Reply:
> Now it is checked if the attribute belongs to a field within per attribute checks

I don't understand how this check can fail. We call fieldDirAttrCheckConsts only on fields, so by definition they all are fields. It is not possible to write a descriptions that will trigger this error, right? If so, I would just remove this function entirely.

Reply:
This is right, [removed](https://github.com/google/syzkaller/pull/2008/commits/50e14bf9892ae773c277c4cfab04dfef256d1803)